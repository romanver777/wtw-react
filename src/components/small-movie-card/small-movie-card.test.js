/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { Router } from "react-router";
import history from "../../history";

import SmallMovieCard from "./small-movie-card";
import films from "../../mocks/films.json";

describe("SmallMovieCard component", () => {
  describe("with init props", () => {
    const props = {
      film: films[2],
      handleMouseEnter: () => {},
      handleMouseLeave: () => {},
      isOpen: false,
      data: null,
    };
    const Trenderer = TestRenderer.create(
      <Router history={history}>
        <SmallMovieCard {...props} />
      </Router>
    );

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });

    it("don't render video elem", () => {
      expect(Trenderer.root.findAllByType("iframe")).toEqual([]);
    });
  });

  describe("with Video props", () => {
    const propsVideo = {
      film: films[2],
      handleMouseEnter: () => {},
      handleMouseLeave: () => {},
      isOpen: true,
      data: {
        items: [
          {
            url: "https://www.youtube.com/watch?v=mccs8Ql8m8o",
            name: "Французский трейлер",
            site: "YOUTUBE",
          },
          {
            url: "https://widgets.kinopoisk.ru/discovery/trailer/13420?onlyPlayer=1&autoplay=1&cover=1",
            name: "Фрагмент (содержит спойлер!) (дублированный)",
            site: "KINOPOISK_WIDGET",
          },
          {
            url: "https://www.youtube.com/v/NRT11KuW6L0",
            name: "Немецкий трейлер",
            site: "YOUTUBE",
          },
        ],
        total: 3,
      },
    };
    const Trenderer = TestRenderer.create(
      <Router history={history}>
        <SmallMovieCard {...propsVideo} />
      </Router>
    );

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });

    it("renders video elem", () => {
      expect(Trenderer.root.findAllByType("iframe")).toHaveLength(1);
    });
  });
});
