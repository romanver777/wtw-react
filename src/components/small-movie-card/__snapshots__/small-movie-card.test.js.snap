// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard component with Video props renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Невероятные (сериал)"
      src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#/film/1245486"
      onClick={[Function]}
    >
      Невероятные (сериал)
    </a>
  </h3>
  <iframe
    allowFullScreen={false}
    frameBorder="0"
    src="https://youtube.com/embed/mccs8Ql8m8o?autoplay=1&mute=1&controls=0"
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 4,
      }
    }
  />
</article>
`;

exports[`SmallMovieCard component with init props renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Невероятные (сериал)"
      src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#/film/1245486"
      onClick={[Function]}
    >
      Невероятные (сериал)
    </a>
  </h3>
</article>
`;
